tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -117,17 +117,24 @@ spec:
             DIGEST="${CONTAINER_IMAGE##*@}"
             PULLSPEC="docker://${REPOSITORY}@${DIGEST}"
             MEDIA_TYPE=$(skopeo inspect --raw "$PULLSPEC" | jq -r .mediaType)
-            skopeo inspect --no-tags "$PULLSPEC" > /tmp/skopeo-inspect.json
+            ARCHS=$(get-image-architectures "$PULLSPEC")
+            for ARCH in $ARCHS; do
+              IMAGE_ARCH=$(skopeo inspect --override-os linux --override-arch "$ARCH" \
```

Comment:
I find the var name confusing. This is not the arch, it's the digest of the image. So I would suggest just `DIGEST` or `IMAGE_DIGEST`.

But more importantly, why do we need need to first get the arch image digest and then do another `skopeo inspect`? You already get the same response with the first one.

So I think you can just do `skopeo inspect --override-os linux --override-arch "$ARCH" "$PULLSPEC" > /tmp/skopeo-inspect-${ARCH}.json` and be done.

Reply:
True. I will fix it.

Reply:
So, just looking it again now - correct me if I am wrong - the thing is that the image sha is different for each arch.
I noticed that by using the `--override-arch` it will override it but the sha will the still the same one, so I assumed that by doing it without getting the right sha, it would be enforcing the arch in the wrong image. 

I need more tests anyway to understand properly this procedure.

Reply:
One quick example:

If you inspect `registry.access.redhat.com/ubi9-minimal@sha256:582e18f13291d7c686ec4e6e92d20b24c62ae0fc72767c46f30a69b1a6198055`

```
RAW_OUTPUT='{
    "manifests": [
        {
            "digest": "sha256:06d06f15f7b641a78f2512c8817cbecaa1bf549488e273f5ac27ff1654ed33f0",
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 429
        },
        {
            "digest": "sha256:199d5daca3dba0a7deaf0086331917dee256089e94272bef5613517d0007f6f5",
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            },
            "size": 429
        },
```

So the digests are different, so I think the proper is getting it for each arch, then enforcing it per each. What maybe we can change to not need to call it again, is adding the digest to the result for `get-image-architectures` - maybe a parameter `--digest` so we don't need to call it multiple times.

Reply:
I think I have a wrong assumption here. The component will be built on top of a single arch, so the image will have a single arch, not a list. I tested with an image that might not reflect the real use case. Trying one of our release-service builds, it has a single arch. So probably I don´t need any loop of archs.

Reply:
You're mixing two things. The snapshot will have the digests for the top level thing - either the image itself, or the manifest list digest in case of multiarch images. In both cases using that digest is enough to get the specific arch image you want with `--override-arch` and ``--override-os` - you don't need to worry about the digest of the arch image itself.

Reply:
> I think I have a wrong assumption here. The component will be built on top of a single arch, so the image will have a single arch, not a list. I tested with an image that might not reflect the real use case. Trying one of our release-service builds, it has a single arch. So probably I don´t need any loop of archs.

The whole point of this PR is to start supporting multiarch images, so I don't understand what you mean by saying that you looked at what we build today and you see single arch images. The thing is that we need to support both single arch and multiarch. But from some discussions in the chat, it seems that in the future, our build system will always produce manifest list images even for single arch.

Reply:
I am confused about what we will "receive" from build-service. For instance  in case `A`

- `component A` is built on top of a `amd64` image and `snapshot-A` is created; the release pipeline runs until create-pyxis-image
- `component A` `snapshot-A` is handled by `create-pyxis-image` 

the question comes here:
will `component A` image  contain only `amd64` information or eventually have multiple archs? even though the component was not built for other archs...

Reply:
btw, found the feature architecture paper. I will take a good read at it so I don't need to flood here anymore.

Reply:
I don't understand the question. Why would we even care how the build system determines what architectures to build? We only care about the result and then we detect what architectures the image has.

Reply:
Unf. @mmalina I am not that smart. Sometimes I need to search/read a lot to understand simple things to reach the point you reach just by looking.

Reply:
Don't worry, I search all the time too 🙂

Reply:
The variable name was updated and also changed the way I was calling skopeo.

Reply:
Thanks, I added more comments, but this is ok.